x/liquidity/simulation/operations.go

```diff
@@ -0,0 +1,439 @@
+		}
+
+		account := ak.GetAccount(ctx, simAccount.Address)
+		spendable := bk.SpendableCoins(ctx, account.GetAddress())
+
+		fees, err := simtypes.RandomFees(r, ctx, spendable)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgDepositToLiquidityPool, "unable to generate fees"), nil, err
+		}
+
+		depositor := account.GetAddress()
+		depositCoins := sdk.NewCoins(randomDepositCoin(r, pool.ReserveCoinDenoms[0]), randomDepositCoin(r, pool.ReserveCoinDenoms[1]))
+
+		msg := types.NewMsgDepositToLiquidityPool(depositor, pool.PoolId, depositCoins)
+
+		txGen := simappparams.MakeTestEncodingConfig().TxConfig
+		tx, err := helpers.GenTx(
+			txGen,
+			[]sdk.Msg{msg},
+			fees,
+			helpers.DefaultGenTxGas,
+			chainID,
+			[]uint64{account.GetAccountNumber()},
+			[]uint64{account.GetSequence()},
+			simAccount.PrivKey,
+		)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "unable to generate mock tx"), nil, err
+		}
+
+		_, _, err = app.Deliver(txGen.TxEncoder(), tx)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "unable to deliver tx"), nil, err
+		}
+
+		return simtypes.NewOperationMsg(msg, true, ""), nil, nil
+	}
+}
+
+// SimulateMsgWithdrawFromLiquidityPool generates a MsgWithdrawFromLiquidityPool with random values
+// nolint: interfacer
+func SimulateMsgWithdrawFromLiquidityPool(ak types.AccountKeeper, bk types.BankKeeper, k keeper.Keeper) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		/*
+			1. Check if there's any liquidity pool created; if there isn't, then return NoOpMsg
+			2. Get any available simulated account and retrieve private key to sign tx message
+			2. Get random liquidity pool and mint pool coin (LP token) to the simulated account
+			3. Withdraw random amounts from the liquidity pool
+		*/
+
+		if len(k.GetAllLiquidityPools(ctx)) == 0 {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgWithdrawFromLiquidityPool, "number of liquidity pools equals zero"), nil, nil
+		}
+
+		pool, ok := randomLiquidity(r, k, ctx)
+		if !ok {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgWithdrawFromLiquidityPool, "unable to pick liquidity pool"), nil, nil
+		}
+
+		// need to retrieve random simulation account to retrieve PrivKey
+		simAccount := accs[simtypes.RandIntBetween(r, 0, len(accs))]
+
+		// if simaccount.PrivKey == nil, delegation address does not exist in accs. Return error
+		if simAccount.PrivKey == nil {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgWithdrawFromLiquidityPool, "account private key is nil"), nil, nil
+		}
+
+		poolCoinDenom := pool.GetPoolCoinDenom()
+
+		// mint pool coin to the simulated account
+		err := mintCoins(r, simAccount.Address, []string{poolCoinDenom}, bk, ctx)
```

Comment:
Minting poolCoin may affect the reservePool share, so it would be better to perform random simAccount only if it is a withdrawer with the poolcoin balance in the previous operation.

Reply:
So the logic should be to check if simAccount has pool coin and then proceed. If simulated account doesn't have and pool coin, then return NoOpMsg.

Reply:
That's right.